app/ante/evm/eth.go

```diff
@@ -337,9 +335,13 @@ func (ctd CanTransferDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 		stateDB := statedb.New(ctx, ctd.evmKeeper, statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes())))
 		evm := ctd.evmKeeper.NewEVM(ctx, coreMsg, cfg, evmtypes.NewNoOpTracer(), stateDB)
 
+		// Sender address should be in the tx cache from the previous AnteHandle call.
+		// From may be either actual from address or the granter address.
+		sender := common.BytesToAddress(msgEthTx.GetFrom())
+
 		// check that caller has enough balance to cover asset transfer for **topmost** call
 		// NOTE: here the gas consumed is from the context with the infinite gas meter
-		if coreMsg.Value().Sign() > 0 && !evm.Context.CanTransfer(stateDB, coreMsg.From(), coreMsg.Value()) {
+		if coreMsg.Value().Sign() > 0 && !evm.Context.CanTransfer(stateDB, sender, coreMsg.Value()) {
```

Comment:
you already replaced the `from` in the core message so imo no need unnecessary change diff like this